tests/unit/test_function.py

```diff
@@ -787,3 +787,107 @@ def test_low_pass_filter(alpha):
             f"The filtered value at index {i} is not the expected value. "
             f"Expected: {expected}, Actual: {filtered_func.source[i][1]}"
         )
+
+
+def test_average_function_ndarray():
+
+    dummy_function = Function(
+        source=[
+            [0, 0],
+            [1, 1],
+            [2, 0],
+            [3, 1],
+            [4, 0],
+            [5, 1],
+            [6, 0],
+            [7, 1],
+            [8, 0],
+            [9, 1],
+        ],
+        inputs=["x"],
+        outputs=["y"],
+    )
+    avg_function = dummy_function.average_function()
+
+    assert isinstance(avg_function, Function)
+    assert np.isclose(avg_function(0), 0)
+    assert np.isclose(avg_function(9), 0.5)
+
+
+def test_average_function_callable():
+
+    dummy_function = Function(lambda x: 2)
+    avg_function = dummy_function.average_function(lower=0)
+
+    assert isinstance(avg_function, Function)
+    assert np.isclose(avg_function(1), 2)
+    assert np.isclose(avg_function(9), 2)
+
+
+@pytest.mark.parametrize(
+    "lower, upper, sampling_frequency, window_size, step_size, remove_dc, only_positive",
+    [
+        (0, 10, 100, 1, 0.5, True, True),
+        (0, 10, 100, 1, 0.5, True, False),
+        (0, 10, 100, 1, 0.5, False, True),
+        (0, 10, 100, 1, 0.5, False, False),
+        (0, 20, 200, 2, 1, True, True),
+    ],
+)
+def test_short_time_fft(
```

Comment:
@Lucas-Prates could yo review this test please? `test_short_time_fft`

Reply:
Looks okay to me! Now, if you want to include a "numeric assertion" test, since you are using pure signal (i.e. sine function), all the result functions should be a Dirac delta function, i.e. a pulse function, at the given frequency, 5 Hz in this case. Something like this

![image](https://github.com/user-attachments/assets/442a104b-ad4c-4b81-a26b-f8f2498b3397)

So, if you want to, you could include an extra argument `true_frequency`, and then

```
signal = np.sin(2 * np.pi * true_frequency * t)
.
.
.
for f in stft_results:
    pytest.approx(1, 0.01) == f[true_frequency - 1][1]
```